tools/xibuild/Main.cs

```diff
@@ -0,0 +1,220 @@
+
+			var SearchPathsOS = IsMacOS ? "osx" : "unix";
+			var projectImportSearchPaths = toolsets.SelectSingleNode ("projectImportSearchPaths");
+			var searchPaths = projectImportSearchPaths.SelectSingleNode ($"searchPaths[@os='{SearchPathsOS}']") as XmlElement;
+
+			//NOTE: on Linux, the searchPaths XML element does not exist, so we have to create it
+			if (searchPaths == null) {
+				searchPaths = dstXml.CreateElement ("searchPaths");
+				searchPaths.SetAttribute ("os", SearchPathsOS);
+
+				var property = dstXml.CreateElement ("property");
+				property.SetAttribute ("name", "MSBuildExtensionsPath");
+				property.SetAttribute ("value", "");
+				searchPaths.AppendChild (property);
+
+				property = dstXml.CreateElement ("property");
+				property.SetAttribute ("name", "MSBuildExtensionsPath32");
+				property.SetAttribute ("value", "");
+				searchPaths.AppendChild (property);
+
+				property = dstXml.CreateElement ("property");
+				property.SetAttribute ("name", "MSBuildExtensionsPath64");
+				property.SetAttribute ("value", "");
+				searchPaths.AppendChild (property);
+
+				projectImportSearchPaths.AppendChild (searchPaths);
+			}
+
+			string monoExternal = IsMacOS ? "/Library/Frameworks/Mono.framework/External/" : "/usr/lib/mono";
+			string [] ProjectImportSearchPaths = new [] { Environment.GetEnvironmentVariable ("MSBuildExtensionsPathFallbackPathsOverride"), Path.Combine (monoExternal, "xbuild") };
+
+			foreach (XmlNode property in searchPaths.SelectNodes ("property[starts-with(@name, 'MSBuildExtensionsPath')]/@value"))
+				property.Value = string.Join (";", ProjectImportSearchPaths);
+
+			SetToolsetProperty ("MSBuildToolsPath", MSBuildBin);
+			SetToolsetProperty ("MSBuildToolsPath32", MSBuildBin);
+			SetToolsetProperty ("MSBuildToolsPath64", MSBuildBin);
+			SetToolsetProperty ("MSBuildExtensionsPath", MSBuildExtensionsPath);
+			SetToolsetProperty ("MSBuildExtensionsPath32", MSBuildExtensionsPath);
+			SetToolsetProperty ("MSBuildExtensionsPath64", MSBuildExtensionsPath);
+			SetToolsetProperty ("RoslynTargetsPath", Path.Combine (MSBuildBin, "Roslyn"));
+			SetToolsetProperty ("TargetFrameworkRootPath", FrameworksDirectory + Path.DirectorySeparatorChar); //NOTE: Must include trailing \
+			SetToolsetProperty ("MSBuildSdksPath", MSBuildSdksPath);
+
+			dstXml.Save (targetConfigFile);
+			return;
+
+			void CopyConfigNode (XmlDocument src, XmlNode dstNode)
+			{
+				var srcConfigNode = src.SelectSingleNode ("configuration");
+				if (srcConfigNode != null && srcConfigNode.HasChildNodes) {
+					srcConfigNode.ChildNodes.OfType<XmlNode> ().ToList ()
+							.ForEach (node => dstNode.AppendChild (dstXml.ImportNode (node, deep: true)));
+				}
+			}
+
+			/// <summary>
+			/// If the value exists, sets value attribute, else creates the element
+			/// </summary>
+			void SetToolsetProperty (string name, string value)
+			{
+				if (string.IsNullOrEmpty (value))
+					return;
```

Comment:
WE check for the value, but what about if the name is null or empty?

Reply:
That would be a bug and IMHO letting the exception be thrown would be good. We could check for null/empty name and explicitly show an error+exit, but not sure if that is worth doing here.